### PR TITLE
Bump prometheus version to latest

### PR DESCRIPTION
**Pull request by annanay25**
Hi Thanos community,

Bumping prometheus version to latest in this PR, as required by Cortex :)

Thanks!

Signed-off-by: Annanay <annanayagarwal@gmail.com>

* [ ] I added CHANGELOG entry for this change.
* [ ] Change is not relevant to the end user.

### CODE REVIEW

**Comment by brancz** [comment]
Do you mind if we wait for https://github.com/prometheus/prometheus/pull/7733 to land and pull that in as well? :) just waiting for CI to go green

**Comment by brancz** [comment]
it's merged :) could you include it?

**Comment by annanay25** [comment]
Thanks @brancz! :)

**Comment by brancz** [comment]
Looks like a couple of CI issues, otherwise lgtm

**Comment by bwplotka** [comment]
Thanks!

**Comment by annanay25** [comment]
Looks like the CI on tests timed out.

**Comment by annanay25** [comment]
Okay I'll get back to digging why the tests are broken in a bit, but is there something obvious I'm missing?

**Comment by annanay25** [comment]
Squashed and updated. 🤞 on CI.

File: pkg/store/tsdb_test.go [hunk 1]
```diff
@@ -203,9 +203,8 @@ func TestTSDBStore_LabelNames(t *testing.T) {
 		expectedNames []string
 	}{
 		{
-			title:         "no label in tsdb",
-			labels:        []string{},
-			expectedNames: []string{},
```
Comment (pracucci) [inline pkg/store/tsdb_test.go:208]: Any reason why this has changed?

File: pkg/receive/multitsdb.go [hunk 2]
```diff
@@ -388,7 +388,7 @@ func (s *ReadyStorage) Querier(ctx context.Context, mint, maxt int64) (storage.Q
 // Appender implements the Storage interface.
 func (s *ReadyStorage) Appender() (storage.Appender, error) {
 	if x := s.get(); x != nil {
-		return x.Appender()
+		return x.Appender(context.TODO())
```
Comment (pracucci) [inline pkg/receive/multitsdb.go:391]: @bwplotka What do you think about propagating the context to `ReadyStorage.Appender()`? I've looked at parent callers and looks an easy thing to do: `Handler.fanoutForward()` > `Writer.Write()` > this function.

Comment (brancz) [reply pkg/receive/multitsdb.go:391]: yes I think passing this makes a lot of sense

Comment (annanay25) [reply pkg/receive/multitsdb.go]: Thanks, addressed.

**Comment by bwplotka** [comment]
e2e:

> === CONT  TestRule
15:30:49 Error: No such network: e2e_test_rule

15:30:49 Unable to remove docker network e2e_test_rule : exit status 1

:thinking: 

retrying

**Comment by bwplotka** [comment]
Something is still not right - build errors

**Comment by annanay25** [comment]
The build is failing with 
```
##[error]pkg/promclient/promclient.go:513:2: invalid case parser.ValueTypeMatrix in switch on m.Data.ResultType (mismatched types parser.ValueType and string)
```
I don't see `parser.ValueTypeMatrix` anywhere - 
https://github.com/thanos-io/thanos/blob/9ef3d48623caf6b42d45a019efc855ed0e1341f6/pkg/promclient/promclient.go#L508-L518

**Comment by yeya24** [comment]
@annanay25 I just added that in pr https://github.com/thanos-io/thanos/pull/2995. Can you please try to rebase master?

**Comment by jtlisi** [comment]
@annanay25 I took a look at your branch. I cherry-picked your commits directly on-top and dug into some of the integration tests issues you are facing. Specifically `rules_test.go`.

I noticed two things. First some of the tests were timing out due to backoffs being hit when checking if metrics exist. By reducing the eval interval of the rules, these errors went away.

I was able to fix this by adding the following to the generated rule groups yaml:

```yaml
  interval: 500ms
```

Next I noticed partial responses being returned which was causing the following rule group not to evaluate:

```yaml
groups:
- name: example_abort
  interval: 500ms
  # Abort should be a default: partial_response_strategy: "ABORT"
  rules:
  - alert: TestAlert_AbortOnPartialResponse
    # It must be based on actual metrics otherwise call to StoreAPI would be not involved.
    expr: absent(some_metric)
    labels:
      severity: page
    annotations:
      summary: "I always complain, but I don't allow partial response in query."
```

I'm not super familiar with the query layer of Thanos or if a partial response is valid in this case. However, by updating the expected alerts to not include the above rule, I was able to get the integration test to pass.

**Comment by annanay25** [comment]
Thanks @jtlisi for the digging! For the last point, I think the test expects the alert to be generated because we received a partial response. I think that should be in place. I'll check what changed in promclient, and update this test.

**Comment by bwplotka** [comment]
I think @pracucci and @pstibrany are using MacOS with e2e just fine :thinking: 

Also I can help as I am blocked with my PR on this one.

**Comment by annanay25** [comment]
@bwplotka - I have a thread in the #thanos slack channel if you're available. :)

**Comment by bwplotka** [comment]
Hey e2e locally returns:


```
    TestRule/attach_query: rule_test.go:395: rule_test.go:395:
        
         unexpected error: unable to find metrics [alertmanager_alerts] with expected values. Last error: <nil>. Last values: [1]
        
    --- FAIL: TestRule/attach_query (45.36s)
```
```
    TestRule/attach_am1: rule_test.go:420: rule_test.go:420:
        
         unexpected error: unable to find metrics [alertmanager_alerts] with expected values. Last error: <nil>. Last values: [1]
        
    --- FAIL: TestRule/attach_am1 (23.52s)
```
```
=== RUN   TestRule/am1_drops_again
    TestRule/am1_drops_again: rule_test.go:449: rule_test.go:449:
        
         unexpected error: unable to find metrics [alertmanager_alerts] with expected values. Last error: <nil>. Last values: [1]
        
    --- FAIL: TestRule/am1_drops_again (22.65s)
```

**Comment by bwplotka** [comment]
On top of that we have race on following test:

```
=== RUN   TestRun
panic: test timed out after 10m0s

goroutine 10 [running]:
testing.(*M).startAlarm.func1()
	/home/bwplotka/.gvm/gos/go1.14.2/src/testing/testing.go:1459 +0xdf
created by time.goFunc
	/home/bwplotka/.gvm/gos/go1.14.2/src/time/sleep.go:168 +0x44

goroutine 1 [chan receive]:
testing.(*T).Run(0xc00014fc20, 0xd5701f, 0x7, 0xd8e9e8, 0x490186)
	/home/bwplotka/.gvm/gos/go1.14.2/src/testing/testing.go:1043 +0x37e
testing.runTests.func1(0xc00014fb00)
	/home/bwplotka/.gvm/gos/go1.14.2/src/testing/testing.go:1284 +0x78
testing.tRunner(0xc00014fb00, 0xc00025fe10)
	/home/bwplotka/.gvm/gos/go1.14.2/src/testing/testing.go:991 +0xdc
testing.runTests(0xc0002621c0, 0x143fd00, 0x7, 0x7, 0x0)
	/home/bwplotka/.gvm/gos/go1.14.2/src/testing/testing.go:1282 +0x2a7
testing.(*M).Run(0xc00011e580, 0x0)
	/home/bwplotka/.gvm/gos/go1.14.2/src/testing/testing.go:1199 +0x15f
main.main()
	_testmain.go:56 +0x135

goroutine 21 [chan receive]:
github.com/prometheus/prometheus/rules.(*Manager).Run(...)
	/home/bwplotka/Repos/thanosgopath/pkg/mod/github.com/prometheus/prometheus@v1.8.2-0.20200805082714-e0cf219f0de2/rules/manager.go:886
github.com/thanos-io/thanos/pkg/rules.(*Manager).Run(0xc00012ac30)
	/home/bwplotka/Repos/thanos/pkg/rules/manager.go:149 +0xb2
github.com/thanos-io/thanos/pkg/rules.TestRun(0xc00014fc20)
	/home/bwplotka/Repos/thanos/pkg/rules/manager_test.go:89 +0x62a
testing.tRunner(0xc00014fc20, 0xd8e9e8)
	/home/bwplotka/.gvm/gos/go1.14.2/src/testing/testing.go:991 +0xdc
created by testing.(*T).Run
	/home/bwplotka/.gvm/gos/go1.14.2/src/testing/testing.go:1042 +0x357

goroutine 23 [select]:
github.com/prometheus/prometheus/rules.(*Group).run(0xc00017e2a0, 0xe8bdc0, 0xc00003e100)
	/home/bwplotka/Repos/thanosgopath/pkg/mod/github.com/prometheus/prometheus@v1.8.2-0.20200805082714-e0cf219f0de2/rules/manager.go:388 +0x5bd
github.com/prometheus/prometheus/rules.(*Manager).Update.func1.1(0xc00012ac80, 0xc00017e2a0)
	/home/bwplotka/Repos/thanosgopath/pkg/mod/github.com/prometheus/prometheus@v1.8.2-0.20200805082714-e0cf219f0de2/rules/manager.go:952 +0x60
created by github.com/prometheus/prometheus/rules.(*Manager).Update.func1
	/home/bwplotka/Repos/thanosgopath/pkg/mod/github.com/prometheus/prometheus@v1.8.2-0.20200805082714-e0cf219f0de2/rules/manager.go:947 +0x56

Process finished with exit code 1
```

I might take a look and fix very first thing in Monday morning or later today depending on my time, unless you manage to do it here first! (:

**Comment by annanay25** [comment]
Just to get this clear, in Thanos do we want to alert at all on `TestAlert_AbortOnPartialResponse`? It seems that the `partial_response_strategy` is set to `ABORT`, and the other option is to `WARN` (Intuitively it sounds like `WARN` should alert but `ABORT` should not, is this correct?

**Comment by annanay25** [comment]
> I see what's the problem, you did not apply change I made here https://github.com/thanos-io/thanos/pull/2876/files#diff-d726ccf140401b6f917e2d90cbbde420L149

> I can port it now if you want


You're right! So sorry. Sure, feel free to port on this PR! 

> Don't get this question. This is alert. We want to have this alert yes. (: Alerts to alert about things. (:

Haha :) Okay just wanted to make sure the `partial_response_strategy` was not for alerting purposes.

**Comment by bwplotka** [comment]
Fixed, I believe this was the reason for two tests (manager and e2e) to not work.

> Haha :) Okay just wanted to make sure the partial_response_strategy was not for alerting purposes.

strategy is to choose if you want to return query error or success if there is some warning (e.g some data backend is unavailable). If you care for consistent data you should use `ABORT`

**Comment by annanay25** [comment]
Thanks for explaining! :)

**Comment by bwplotka** [comment]
Easy thing to fix left: https://app.circleci.com/pipelines/github/thanos-io/thanos/2719/workflows/50b01690-9094-410f-874b-4632c0753799/jobs/10945

**Comment by bwplotka** [comment]
```
--- FAIL: TestTSDBStore_LabelNames (0.02s)
    --- FAIL: TestTSDBStore_LabelNames/query_range_outside_tsdb_head (0.00s)
        tsdb_test.go:279: tsdb_test.go:279:
            
            	exp: []string{}
            
            	got: []string(nil)
            
            Diff:
            --- Expected
            +++ Actual
            @@ -1,3 +1,2 @@
            -([]string) {
            -}
            +([]string) <nil>
             
            
            
--- FAIL: TestTSDBStore_LabelValues (0.01s)
    --- FAIL: TestTSDBStore_LabelValues/query_time_range_outside_head (0.00s)
        tsdb_test.go:379: tsdb_test.go:379:
            
            	exp: []string{}
            
            	got: []string(nil)
            
            Diff:
            --- Expected
            +++ Actual
            @@ -1,3 +1,2 @@
            -([]string) {
            -}
            +([]string) <nil>
             
```
Stil to fix

**Comment by bwplotka** [comment]
I pushed fix

**Comment by bwplotka** [comment]
Thanks! @annanay25

**Comment by annanay25** [comment]
I fixed test and then merged upstream 🤦 Thanks for the fix!

**Pull request merged**